src/components/molecules/Visualizer/Plugin/types.ts

```diff
@@ -74,6 +74,8 @@ export type Layers = {
   ) => T | undefined;
   readonly isLayer: (obj: any) => obj is Layer;
   readonly overrideProperty: (id: string, property: any) => void;
+  readonly append: (layer: Layer, parentId?: string) => string;
+  readonly extensionIds: string[];
```

Comment:
- Currently the `append` method accept only one layer and its parentId as option so it just do append to root layer or parent layer. But i think it should support tree structure as it was designed. I'm going to update this part and if it works update the method name to `add`.
- The `getExtensionIds` API comes from the former talk. I agree that this can be done in document and i'm going to remove it.